deploy/util.go

```diff
@@ -0,0 +1,205 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/state"
+	"github.com/nspcc-dev/neo-go/pkg/encoding/address"
+	"github.com/nspcc-dev/neo-go/pkg/neorpc"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/invoker"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neofs-contract/common"
+	"go.uber.org/zap"
+)
+
+func isErrContractAlreadyUpdated(err error) bool {
+	return strings.Contains(err.Error(), common.ErrAlreadyUpdated)
+}
+
+func isErrTLDNotFound(err error) bool {
+	return strings.Contains(err.Error(), "TLD not found")
+}
+
+// blockchainMonitor is a thin utility around Blockchain providing state
+// monitoring.
+type blockchainMonitor struct {
+	logger *zap.Logger
+
+	blockchain Blockchain
+
+	blockInterval time.Duration
+
+	height atomic.Uint32
+
+	chConnLost chan struct{}
+}
+
+// newBlockchainMonitor constructs and runs monitor for the given Blockchain.
+func newBlockchainMonitor(l *zap.Logger, b Blockchain, chNewBlock chan<- struct{}) (*blockchainMonitor, error) {
```

Comment:
Have you completely removed this monitor from NeoFS?

Reply:
Looks like yes. It wasn't used in any another place